src/Main.js

```diff
@@ -18,52 +20,100 @@ WebhookHandler.on('error', (err) => {
-    const pr = ev.payload.pull_request;
-    Logger.info("pull_request_review event:", ev.payload.id, pr.number, pr.head.sha, pr.state);
-    Merger.run();
-});
+WebhookHandler.on('pull_request_review', HandlerWrap((ev) => {
+        const pr = ev.payload.pull_request;
+        Logger.info("pull_request_review event:", ev.payload.id, pr.number, pr.head.sha, pr.state);
+        return Util.PrId.PrNum(pr.number);
+    })
+);
 
 // https://developer.github.com/v3/activity/events/types/#pullrequestevent
-WebhookHandler.on('pull_request', (ev) => {
-    const pr = ev.payload.pull_request;
-    Logger.info("pull_request event:", ev.payload.id, pr.number, pr.head.sha, pr.state);
-    Merger.run();
-});
+WebhookHandler.on('pull_request', HandlerWrap((ev) => {
+        const pr = ev.payload.pull_request;
+        Logger.info("pull_request event:", ev.payload.id, pr.number, pr.head.sha, pr.state);
+        return Util.PrId.PrNum(pr.number);
+    })
+);
 
 // https://developer.github.com/v3/activity/events/types/#statusevent
-WebhookHandler.on('status', (ev) => {
-    const e = ev.payload;
-    Logger.info("status event:", e.id, e.sha, e.context, e.state);
-    Merger.run();
-});
+WebhookHandler.on('status', HandlerWrap((ev) => {
+        const e = ev.payload;
+        Logger.info("status event:", e.id, e.sha, e.context, e.state);
+        const branches = Array.from(e.branches, b => b.name);
+        return Util.PrId.BranchList(branches, e.commit.commit.message);
+    })
+);
 
 // https://developer.github.com/v3/activity/events/types/#pushevent
-WebhookHandler.on('push', (ev) => {
-    const e = ev.payload;
-    Logger.info("push event:", e.ref);
-    Merger.run();
-});
+WebhookHandler.on('push', HandlerWrap((ev) => {
+        const e = ev.payload;
+        Logger.info("push event:", e.ref);
+
+        // e.ref as refs/heads/branch_name
+        const parts = e.ref.split('/');
+        const branch = parts[parts.length-1];
+        return Util.PrId.BranchList([branch], e.head_commit ? e.head_commit.message : null);
+    })
+);
 
 // https://docs.github.com/ru/webhooks/webhook-events-and-payloads#workflow_run
-WebhookHandler.on('workflow_run', (ev) => {
-    const e = ev.payload.workflow_run;
-    Logger.info("workflow_run event:", e.head_sha);
-    Merger.run();
-});
+WebhookHandler.on('workflow_run', HandlerWrap((ev) => {
+        const e = ev.payload.workflow_run;
+        Logger.info("workflow_run event:", e.head_sha);
+        // e.pull_requests is empty for the staged commit
+        if (e.head_branch === Config.stagingBranch()) {
+            assert(!e.pull_requests.length);
+            return Util.PrId.Sha(e.head_sha);
+        }
+        if (!e.pull_requests.length) {
+            // e.pull_requests is empty, e.g., for master commits
+            Logger.info("workflow_run event: no PR for ", e.head_sha);
+            return [];
+        }
+        const list = Array.from(e.pull_requests, v => v.number);
+        return Util.PrId.PrNumList(list);
```

Comment:
I suspect the code becomes simpler or more readable if this snippet is moved to the top of the handler, protected by the `e.pull_requests.length` condition: If GitHub tells us associated PRs, use that info. Otherwise, handle the special case of a staged branch. Please try that processing order.